cats/src/main/scala/magnolify/cats/auto/CatsMacros.scala

```diff
@@ -29,6 +29,12 @@ private object CatsMacros {
     q"""_root_.magnolify.cats.semiauto.EqDerivation.apply[$wtt]"""
   }
 
+  def genHashMacro[T: c.WeakTypeTag](c: whitebox.Context): c.Tree = {
```

Comment:
I am unaware of this, but why do we need this redirection? We aren't cleaning up annotations or anything like that, right?

Reply:
Because the `implicit def genHash[T]: Hash[T]` etc. in `magnolify.cats.auto._` need to pass on `WeakTypeTag` and only macros can do that.